#!/usr/bin/env python

PACKAGE = "dcsl_low_level_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

robot_enum = gen.enum([ gen.const("None", int_t, -1, "Don't control a robot"),
                        gen.const("Robot0", int_t, 0, "Control Robot 0 with joystick"),
                        gen.const("Robot1", int_t, 1, "Control Robot 1 with joystick"),
                        gen.const("Robot2", int_t, 2, "Control Robot 2 with joystick"),
                        gen.const("Robot3", int_t, 3, "Control Robot 3 with joystick")],
                      "An enum to set which robot is controlled with the joystick")

gen.add("JoystickRobot", int_t, 0, "Which robot to control with joystick", -1, -1, 3, edit_method=robot_enum)

exit(gen.generate(PACKAGE, "dcsl_beluga_low_level_control.py", "dcsl_beluga_low_level_control_config"))
